src/idsk/components/table-filter/table-filter.js

```diff
@@ -0,0 +1,235 @@
+import '../../../govuk/vendor/polyfills/Event' // addEventListener and event.target normaliziation
+import {toggleClass} from "../../common";
+
+
+function TableFilter($module) {
+  this.$module = $module
+  this.selectedFitlersCount = 0
+  this.$activeFilters = []
+}
+
+TableFilter.prototype.init = function () {
+  // Check for module
+  var $module = this.$module
+  if (!$module)
+    return
+
+  // toggle for showing content
+  var $toggleButtons = $module.querySelectorAll('.idsk-filter-menu__toggle')
+  if (!$toggleButtons)
+    return
+
+  // submit all filters
+  var $submitFilters = $module.querySelectorAll('.submit-table-filter')
+  if (!$submitFilters)
+    return
+
+  // all inputs for count of selected filters
+  var $filterInputs = $module.querySelectorAll('.govuk-input, .govuk-select')
+  if (!$filterInputs)
+    return
+
+  $toggleButtons.forEach(function ($button) {
+    $button.addEventListener('click', this.handleClickTogglePanel.bind(this))
+  }.bind(this))
+
+  $submitFilters.forEach(function ($button) {
+    $button.addEventListener('click', this.handleSubmitFilter.bind(this))
+  }.bind(this))
+
+  $filterInputs.forEach(function ($input) {
+    // for selects
+    $input.addEventListener('change', this.handleFilterValueChange.bind(this))
+    // for text inputs
+    $input.addEventListener('keyup', function (e) {
+      this.handleFilterValueChange(e)
+
+      // submit if key is enter
+      if (e.key === 'Enter')
+        this.handleSubmitFilter(this)
+    }.bind(this));
+  }.bind(this))
+}
+
+/**
+ * An event handler for click event on $togglePanel - collapse or expand table-filter
+ * @param {object} e
+ */
+TableFilter.prototype.handleClickTogglePanel = function (e) {
+  var $el = e.target || e.srcElement
+  var $expandablePanel = $el.parentNode
+  var $content = $el.nextElementSibling
+
+  // if panel is category, change size of whole panel with animation
+  if ($expandablePanel.classList.contains("idsk-table-filter__category")) {
+    var $categoryParent = $expandablePanel.parentNode
+
+    // made more fluid animations for changed spacing with transition
+    var marginBottomTitle = 20
+    var marginBottomExpandedCategory = 32
+    var newParentHeight = ($content.style.height && $content.style.height !== "0px")
+      ? parseInt($categoryParent.style.height) - $content.scrollHeight - marginBottomTitle + marginBottomExpandedCategory
+      : parseInt($categoryParent.style.height) + $content.scrollHeight + marginBottomTitle - marginBottomExpandedCategory
+
+    $categoryParent.style.height = newParentHeight + "px"
+  }
+
+  // show element after toggle with slide down animation
+  toggleClass($expandablePanel, 'idsk-table-filter--expanded')
+  $content.style.height = ($content.style.height && $content.style.height !== "0px" ? "0" : $content.scrollHeight) + "px";
+
+  // set text for toggle
+  $el.innerHTML = ($content.style.height === "0px" ? "Rozbaliť" : "Zbaliť") + " filter"
```

Comment:
Texty do jska by som nedaval.. v pripade viacjazycnej mutacie je js "nemozne" prelozit. Cekni si ako to je robene v inych komponentoch cez data-* atributy